centralizedmetrics/metrics.go

```diff
@@ -0,0 +1,139 @@
+package centralizedmetrics
+
+import (
+	"database/sql"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/status-im/status-go/centralizedmetrics/common"
+	"github.com/status-im/status-go/centralizedmetrics/providers"
+)
+
+var defaultPollInterval = 10 * time.Second
+
+type MetricsInfo struct {
+	Enabled       bool `json:"enabled"`
+	UserConfirmed bool `json:"userConfirmed"`
+}
+
+type MetricRepository interface {
+	Poll() ([]common.Metric, error)
+	Delete(metrics []common.Metric) error
+	Add(metric common.Metric) error
+	Info() (*MetricsInfo, error)
+	ToggleEnabled(enabled bool) error
+}
+
+type MetricService struct {
+	repository MetricRepository
+	processor  common.MetricProcessor
+	ticker     *time.Ticker
+	done       chan bool
+	started    bool
+	wg         sync.WaitGroup
+}
+
+func NewDefaultMetricService(db *sql.DB) *MetricService {
+	repository := NewSQLiteMetricRepository(db)
+	processor := providers.NewMixpanelMetricProcessor(providers.MixpanelAppID, providers.MixpanelToken, providers.MixpanelBaseURL)
+	return NewMetricService(repository, processor, defaultPollInterval)
+}
+
+func NewMetricService(repository MetricRepository, processor common.MetricProcessor, interval time.Duration) *MetricService {
+	return &MetricService{
+		repository: repository,
+		processor:  processor,
+		ticker:     time.NewTicker(interval),
+		done:       make(chan bool),
+	}
+}
+
+func (s *MetricService) Start() {
+	if s.started {
+		return
+	}
+	s.wg.Add(1)
+	s.started = true
+	go func() {
+		defer s.wg.Done()
+		for {
+			select {
+			case <-s.done:
+				return
+			case <-s.ticker.C:
+				s.processMetrics()
+			}
+		}
+	}()
+}
+
+func (s *MetricService) Stop() {
+	if !s.started {
+		return
+	}
+	s.ticker.Stop()
+	s.done <- true
+	s.wg.Wait()
+	s.started = false
+}
+
+func (s *MetricService) EnsureStarted() error {
+	info, err := s.Info()
+	if err != nil {
+		return err
+	}
+	if info.Enabled {
+		s.Start()
+	}
+	return nil
+}
+
+func (s *MetricService) Info() (*MetricsInfo, error) {
+	return s.repository.Info()
+}
+
+func (s *MetricService) ToggleEnabled(enabled bool) error {
+	err := s.repository.ToggleEnabled(enabled)
```

Comment:
I don't know how consistent we are in status-go, but many places in the repo use the `Is*` prefix on booleans, so the name could be `ToggleIsEnabled(isEnabled)`. But maybe many others don't follow this convention.